xdsl/parser/attribute_parser.py

```diff
@@ -184,24 +184,40 @@ def _parse_dialect_type_or_attribute_inner(
         The contents will be parsed by a user-defined parser, or by a generic parser
         if the dialect attribute/type is not registered.
         """
+        pretty = "." in attr_name
+        if not pretty:
+            self.parse_punctuation("<")
+            attr_name += (
+                "."
+                + self._parse_token(
+                    Token.Kind.BARE_IDENT, "Expected attribute name."
+                ).text
+            )
         attr_def = self.ctx.get_optional_attr(
             attr_name,
             create_unregistered_as_type=is_type,
         )
         if attr_def is None:
             self.raise_error(f"'{attr_name}' is not registered")
 
+        attr = None
         # Pass the task of parsing parameters on to the attribute/type definition
         if issubclass(attr_def, UnregisteredAttr):
             body = self._parse_unregistered_attr_body()
```

Comment:
Will this work when we print back the opaque unregistered attribute?
How do we know how to print it back with the opaque form?

Reply:
Good point! I'll add some logic for that.